lib/thread.c

```diff
@@ -316,11 +317,18 @@ static void show_thread_poll_helper(struct vty *vty, struct thread_master *m)
 	vty_out(vty, "----------------------%s\n", underline);
 	vty_out(vty, "Count: %u/%d\n", (uint32_t)m->handler.pfdcount,
 		m->fd_limit);
-	for (i = 0; i < m->handler.pfdcount; i++)
-		vty_out(vty, "\t%6d fd:%6d events:%2d revents:%2d\n", i,
-			m->handler.pfds[i].fd,
-			m->handler.pfds[i].events,
-			m->handler.pfds[i].revents);
+	for (i = 0; i < m->handler.pfdcount; i++) {
+		if (m->handler.pfds[i].events & POLLIN)
+			thread = m->read[m->handler.pfds[i].fd];
+		else
+			thread = m->write[m->handler.pfds[i].fd];
```

Comment:
It's valid for both read and write to be set (and neither, I think?), and our problems have been with POLLOUT so we definitely need that printed even if POLLIN is set too.

Reply:
yeah I can do that.